pkg/tracker/balance.go

```diff
@@ -43,10 +50,11 @@ func (b *BalanceTracker) Exec(ctx context.Context) error {
 	balance, err := client.BalanceAt(ctx, fromAddress, nil)
 
 	if err != nil {
-		fmt.Println("balance Error, balance.go")
+		level.Error(b.logger).Log("msg", "getting balance", "err", err)
 		return err
```

Comment:
In places where the functions returns the error and stop executing, don't need to log, just pass the error to the caller and let the caller decide whether to log it or pass it up the stack. This way we can even avoid the need to have a logger in functions that just need to return an error. There are few places like this.

Actually in the trackers I think only the gas tracker needs to log as when it fails to get the price from the gasStation website it can get it from the eth client. In these cases where the execution after an error can continue it is ok to print a log otherwise in functions that stop the execution on every error need to just return the error to the caller and no need to print a log.

This way you can see that `CurrentVariablesTracker`, `TributeTracker` don't even need a logger because they just stop the execution on an error and return the error to the caller and the caller can decide whether to print a log or to pass the error up the stack.

You can leave the constructors though, I am sure we will need a constructor at some point anyway.

```suggestion
		return errors.Wrap(err, "getting balance")
```